### PR TITLE
Carry #22228 : Rename graph drivers to storage drivers

### PR DESCRIPTION
**Pull request by vdemeester**
Carry PR #22228 🐸.
Resolving conflicts wasn't easy, let's see if my missed something or not :angel: 

Closes #22228 

---

**\- What I did**

I've renamed the occurrences of `GraphDriver` by `StorageDriver` in the code base to be more consistent with the [docs](https://docs.docker.com/engine/userguide/storagedriver/selectadriver/).

The only externally-visible change of that PR is a direct API translation of the [experimental storage driver plugins API](https://github.com/docker/docker/blob/master/experimental/plugins_graphdriver.md)

**\- How I did it**

I've renamed things via grep/replace but I tried to do that carefully, so variable names and comments are also kept consistent.

**\- How to verify it**

I've run `make` and `make test` and it seems to work. I need to update https://github.com/docker/go-plugins-helpers/pull/42 to use the new StorageDriver naming scheme, I'll do that once/if that PR is merged.

I've tried to run the integration tests for the CLI. The experimental storage driver plugins seem broken (see below).

**\- A picture of a cute animal (not mandatory but encouraged)**

cute camel

### CODE REVIEW

**Comment by vdemeester** [comment]
Already LGTM from @tiborvass (see original PR)
/cc @thaJeztah @runcom @justincormack

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 .PHONY: all binary build build-gccgo cross default docs docs-build docs-shell shell gccgo test test-docker-py test-integration-cli test-unit validate help
 
-# set the graph driver as the current graphdriver if not set
+# set the storage driver as the current graphdriver if not set
```
Comment (justincormack) [inline Makefile]: er "graphdriver" also in this sentence

File: Makefile [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
 DOCKER_GRAPHDRIVER := $(if $(DOCKER_GRAPHDRIVER),$(DOCKER_GRAPHDRIVER),$(shell docker info 2>&1 | grep "Storage Driver" | sed 's/.*: //'))
```
Comment (justincormack) [inline Makefile]: "`DOCKER_GRAPHDRIVER`" here too....

File: project/RELEASE-CHECKLIST.md [hunk 3]
```diff
@@ -215,7 +215,7 @@ open the PR against the "release" branch instead of accidentally against
 
 ### 7. Build release candidate rpms and debs
 
-**NOTE**: It will be a lot faster if you pass a different graphdriver with
+**NOTE**: It will be a lot faster if you pass a different storage driver with
 `DOCKER_GRAPHDRIVER` than `vfs`.
```
Comment (justincormack) [inline project/RELEASE-CHECKLIST.md]: `DOCKER_GRAPHDRIVER` also here, I guess this has not been changed?

File: Makefile (hunk 2 above)
Comment (samoht) [inline Makefile]: this is used by Jenkins config so a bit more hassle to change.

Comment (justincormack) [inline Makefile]: hmm, ok that does make it more complicated....

Comment (vdemeester) [inline Makefile]: ah damn :scream:

**Comment by thaJeztah** [comment]
Looks like we're running into the same failure again on WindowsTP5, so wondering if there is a relation to this change;

```
14:43:46 ---> Making bundle: .ensure-frozen-images-windows (in bundles/1.12.0-dev/test-integration-cli)
14:43:46 ERROR: Could not find windowsservercore images
```

@jhowardmsft any ideas? I already started looking into vendored code, e.g. https://github.com/docker/docker/blob/3d13fddd2bc4d679f0eaa68b0be877e5a816ad53/vendor/src/github.com/Microsoft/hcsshim/layerutils.go#L14-L23, but couldn't find anything really

File: daemon/storage/windows/windows.go [hunk 4]
```diff
@@ -20,27 +20,27 @@ import (
 	"time"
 	"unsafe"
 
-	"github.com/Microsoft/go-winio"
+	winio "github.com/Microsoft/go-winio"
```
Comment (jhowardmsft) [inline daemon/storage/windows/windows.go]: Looking to see anything obvious causing TP5 to fail. This isn't it (I don't think), but why aliased?

**Comment by jhowardmsft** [comment]
@thaJeztah The only thing I can spot is the go-winio aliasing. Everything appears fine.

**Comment by thaJeztah** [comment]
@jhowardmsft thanks for looking. I was wondering if somehow, somewhere something relied on a name. Just weird. Let me just try to run it again, maybe we've just been very unlucky

File: daemon/storage/windows/windows.go (hunk 4 above)
Comment (vdemeester) [inline daemon/storage/windows/windows.go]: hum, it shouldn't matter, the package name is `winio` anyway (https://github.com/Microsoft/go-winio/blob/master/file.go, …)

**Comment by vdemeester** [comment]
Hum `windowsTP5` is definitely not happy :sweat_smile:

**Comment by jhowardmsft** [comment]
Went back to look at this failure after redeploying a bunch of TP5 nodes this evening. Unfortunately it now needs a rebase :cry:

**Comment by vdemeester** [comment]
@jhowardmsft rebased :angel:

**Pull request closed without merging**